python/ray/util/sgd/torch/examples/image-models/train.py

```diff
@@ -0,0 +1,164 @@
+# Based on work by Ross Wightman as part of the timm package
+# (see LICENSE_THIRDPARTY)
+#
+# As modified by
+# - Maksim Smolin in 2020
+
+# Note: other authors MUST include themselves in the above copyright notice
+#       in order to abide by the terms of the Apache license
+
+from os.path import join
+
+from tqdm import trange
+
+import torch.nn as nn
+
+from timm.data import Dataset, create_loader
+from timm.data import resolve_data_config, FastCollateMixup
+from timm.models import create_model, convert_splitbn_model
+from timm.optim import create_optimizer
+from timm.utils import setup_default_logging
+
+import ray
+from ray.util.sgd.utils import BATCH_SIZE
+
+from ray.util.sgd import TorchTrainer
+# from ray.util.sgd.torch import TrainingOperator
+
+from args import parse_args
+
+
+class Namespace(dict):
+    def __init__(self):
+        pass
+
+    def __getattr__(self, attr):
+        if attr not in self:
+            raise AttributeError(attr)
+
+        return self[attr]
+
+    def __setattr__(self, attr, value):
+        self[attr] = value
```

Comment:
is this used anywhere?

Reply:
Oh, not anymore, I used it before I had argument parsing implemented because some of `timm` functions rely on a arparse namespace-like thing

Reply:
Removed it